Makefile

```diff
@@ -24,6 +24,9 @@ develop: requirements  ## install dependencies and build library
 build:  ## build the library
 	python setup.py build build_ext --inplace -- -- -j$(NPROC)
 
+build-debug:  ## build the library ( DEBUG ) - May need a make clean when switching from regular build to build-debug and vice versa
```

Comment:
might want to add to the `.PHONY` block somewhere below (i think), not strictly necessary unless we have a `build-debug` folder at somepoint